src/scmrepo/git/lfs/client.py

```diff
@@ -228,33 +227,48 @@ def __init__(self, url: str, host: str, port: int, path: str):
         super().__init__(url)
         self.host = host
         self.port = port
+        self.username = username
         self.path = path
         self._ssh = _get_ssh_vendor()
 
     @classmethod
     def from_git_url(cls, git_url: str) -> "_SSHLFSClient":
-        result = cls._URL_PATTERN.match(git_url)
-        if not result:
+        if scp_match := SCP_REGEX.match(git_url):
+            # Add an ssh:// prefix and replace the ':' with a '/'.
+            git_url = scp_match.expand("ssh://\1\2/\3")
```

Comment:
I went with converting scp-style urls to `ssh://` urls as done in pip. After that, we parse using `urlparse` to extract `hostname`/`port`/`username`.

https://github.com/pypa/pip/blob/303fed36c1771de4063063a866776a9103972317/src/pip/_internal/vcs/git.py#L402-L405

Reply:
Ah, that's great! :ok_hand: TIL [`Match.expand`](https://docs.python.org/3/library/re.html#re.Match.expand). :smile: And the Walrus operator is very elegant, too. :star_struck: